beacon-chain/powchain/block_reader_test.go

```diff
@@ -185,23 +185,90 @@ func TestBlockExists_UsesCachedBlockInfo(t *testing.T) {
 	require.Equal(t, 0, height.Cmp(header.Number))
 }
 
-func TestBlockNumberByTimestamp(t *testing.T) {
+func TestService_BlockNumberByTimestamp(t *testing.T) {
 	beaconDB, _ := dbutil.SetupDB(t)
+	testAcc, err := contracts.Setup()
+	require.NoError(t, err, "Unable to set up simulated backend")
 	web3Service, err := NewService(context.Background(), &Web3ServiceConfig{
 		HTTPEndPoint: endpoint,
 		BeaconDB:     beaconDB,
 	})
 	require.NoError(t, err)
 	web3Service = setDefaultMocks(web3Service)
+	web3Service.eth1DataFetcher = &goodFetcher{backend: testAcc.Backend}
 
+	for i := 0; i < 200; i++ {
+		testAcc.Backend.Commit()
+	}
 	ctx := context.Background()
-	bn, err := web3Service.BlockNumberByTimestamp(ctx, 150000 /* time */)
+	hd, err := testAcc.Backend.HeaderByNumber(ctx, nil)
+	require.NoError(t, err)
+	web3Service.latestEth1Data.BlockTime = hd.Time
+	bn, err := web3Service.BlockNumberByTimestamp(ctx, 1000 /* time */)
 	require.NoError(t, err)
 	if bn.Cmp(big.NewInt(0)) == 0 {
 		t.Error("Returned a block with zero number, expected to be non zero")
 	}
 }
 
+func TestService_BlockNumberByTimestampLessTargetTime(t *testing.T) {
+	beaconDB, _ := dbutil.SetupDB(t)
+	testAcc, err := contracts.Setup()
+	require.NoError(t, err, "Unable to set up simulated backend")
+	web3Service, err := NewService(context.Background(), &Web3ServiceConfig{
+		HTTPEndPoint: endpoint,
+		BeaconDB:     beaconDB,
+	})
+	require.NoError(t, err)
+	web3Service = setDefaultMocks(web3Service)
+	web3Service.eth1DataFetcher = &goodFetcher{backend: testAcc.Backend}
+
+	for i := 0; i < 200; i++ {
+		testAcc.Backend.Commit()
+	}
+	ctx := context.Background()
+	hd, err := testAcc.Backend.HeaderByNumber(ctx, nil)
+	require.NoError(t, err)
+	web3Service.latestEth1Data.BlockTime = hd.Time
+	// Provide an unattainable target time
+	_, err = web3Service.findLessTargetEth1Block(ctx, hd.Number, hd.Time/2)
+	require.ErrorContains(t, errSearch.Error(), err)
+
+	// Provide an attainable target time
+	bn, err := web3Service.findLessTargetEth1Block(ctx, hd.Number, hd.Time-5)
+	require.NoError(t, err)
+	require.NotEqual(t, hd.Number.Uint64(), bn.Uint64(), "retrieved block is less than the head")
```

Comment:
The error message is incorrect. It should be `retrieved block is not less than the head`.